### DISCUSSION TITLE
Does creating an account in the console trigger `before_create_account` hook?

### DISCUSSION THREAD

**Discussion by whysthatso** [janko/rodauth-rails#277] [Q&A] [answered]
Account.create email: "email@example.com" etc... only creates the account, successfully, but does not seem to call that hook.

**Comment by janko** [comment, marked as answer]
It doesn't. It skips the account creation lifecycle, and only uses Rodauth to hash the password. If you want the hook to be called, you should use the `internal_request` feature, concretely the `create_account` class method it provides.

```rb
RodauthApp.rodauth.create_account(login: "email@example.com", password: "secret123")
```

**Comment by whysthatso** [comment]
dude, you're too fast :)
